src/main/java/net/sf/jabref/exporter/BibDatabaseWriter.java

```diff
@@ -354,6 +356,27 @@ private void writeString(Writer fw, BibtexString bs, Map<String, BibtexString> r
         fw.write("}" + Globals.NEWLINE);
     }
 
+    private void writeUserCommentsForString(BibtexString string, Writer out) throws IOException {
+        String parsedSerialization = string.getParsedSerialization();
+
+        if(parsedSerialization != null) {
+
+            try {
+                // get the text before the string
+                String prolog = string.getParsedSerialization().substring(0, string.getParsedSerialization().indexOf('@'));
```

Comment:
Idea: Move most of the code which determines the comment to `BibEntry.getUserComments` since in my opinion it is the responsibility of the BibEntry to store all the data in an accessible format.